components/layout_2020/positioned.rs

```diff
@@ -846,9 +846,14 @@ impl<'a> AbsoluteAxisSolver<'a> {
                 let stretch_size = free_space -
                     self.computed_margin_start.auto_is(Au::zero) -
                     self.computed_margin_end.auto_is(Au::zero);
-                let used_size = solve_size(Size::Stretch, stretch_size)
-                    .to_definite()
-                    .unwrap();
+                let used_size = match self.alignment.value() {
+                    AlignFlags::STRETCH | AlignFlags::NORMAL => {
+                        solve_size(Size::Stretch, stretch_size)
+                    },
+                    _ => solve_size(Size::FitContent, stretch_size),
+                }
+                .to_definite()
+                .unwrap_or_default();
                 free_space -= used_size;
```

Comment:
Use the `to_definite()` just for this. If it's not definite, I don't think the margins matter at all, but maybe setting `free_space` to zero makes the most sense?

```suggestion
                if let Some(used_size) = size.to_definite() {
                    free_space -= used_size;
                } else {
                    free_space = Au::zero(),
                }
```